### ISSUE TITLE
Stop generating if more than X chunks are currently loaded

### ISSUE THREAD

**Issue by masmc05** [pop4959/Chunky#303]
An addition to <https://github.com/pop4959/Chunky/issues/237>, if there are more than the specified number of chunks currently loaded then block the generation until time passes and the number of chunks will be lower.

Since for most people this may be useless and may ignore this option, you could add this to the command and task, not hide into a command line option.

**Comment by masmc05** [comment]
> Have you tried reducing the duration for ticking? Setting it to a bad value will cause issues, which is one of the reasons I opted to put it behind a flag rather than in plain sight in the config.

yeah, it's a possible fix, but if we want to ensure that chunks are properly ticked (and yes, it had big performance impact, before on the first day we could only take 130 players per server, with this we were able to get 220 players on 20 tps on the first day) but at the same time we want high generation speed, we'll end up in a long time choosing perfect balance between paper generation worker threads and tick time for chunks

**Comment by pop4959** [comment]
I'm glad to hear it makes a difference. And yeah, I'd recommend probably not setting a value higher than 20 or 30 seconds. Need not be 100% perfect, as settling 99% of liquids and such and leaving the remaining 1% should be fine.

Alternatively, you can also do several runs of chunky at lower durations (which in theory also limits simultaneous chunks).

I'm going to consider this fixed / answered for now, but feel free to drop by the [Discord server](https://discord.gg/ZwVJukcNQG) in the chunky support channel if you want to discuss this more. I'm skeptical of trying to do something like limiting loaded chunks as that sounds like it can cause more problems, but perhaps another solution would be acceptable.

**Comment by pop4959** [comment]
e398cc2d8e75f4525c7eee659d04716438e59e5d adds a new flag which can be used in addition to the existing working count flag to effectively limit this in a way that basically achieves the same job as what this suggested, although less jank.
